From 71fc07a2d0af7a1bdd48e2970cd24d2d08f9c782 Mon Sep 17 00:00:00 2001
From: sooti <sooti85@gmail.com>
Date: Sun, 1 Aug 2021 09:55:31 +0300
Subject: [PATCH 35/80] Add prop tweaks for 9 pro/s/Pro Max

* Add audio tweaks
* Use vendor media camera
* Brightness tweak (low brightnness)
---
 phh-on-data.sh | 17 +++++++++++++++++
 rw-system.sh   | 13 +++++++++++--
 2 files changed, 28 insertions(+), 2 deletions(-)

diff --git a/phh-on-data.sh b/phh-on-data.sh
index bc977f0..16c3812 100644
--- a/phh-on-data.sh
+++ b/phh-on-data.sh
@@ -20,6 +20,23 @@ if getprop persist.sys.phh.caf.media_profile |grep -q true; then
     setprop media.settings.xml "/vendor/etc/media_profiles_vendor.xml"
 fi
 
+# Add persist props for Redmi Note 9S/Pro/Max
+if getprop ro.vendor.build.fingerprint |grep -iq -e redmi/curtana \
+  -e redmi/joyeuse -e redmi/excalibur; then
+    setprop persist.sys.phh.disable_audio_effects 1
+    setprop persist.sys.phh.disable_a2dp_offload true
+    setprop persist.bluetooth.bluetooth_audio_hal.disabled true
+    setprop persist.sys.phh.caf.audio_policy 1
+    mount -o bind /system/etc/mixer_paths_wcd937x.xml /vendor/etc/mixer_paths_wcd937x.xml
+    mount -o bind /system/etc/media_profiles_vendor.xml /vendor/etc/media_profiles_vendor.xml
+    setprop ctl.restart vendor.audio-hal-2-0
+    setprop persist.sys.phh.linear_brightness false
+    setprop persist.sys.phh.backlight.scale 1
+    setprop persist.sys.phh.caf.media_profile true
+    setprop persist.sys.overlay.minimal_brightness true
+    pkill -f com.android.bluetooth
+fi
+
 minijailSrc=/system/system_ext/apex/com.android.vndk.v28/lib/libminijail.so
 minijailSrc64=/system/system_ext/apex/com.android.vndk.v28/lib64/libminijail.so
 if [ "$vndk" = 27 ]; then
diff --git a/rw-system.sh b/rw-system.sh
index 0ecc4d0..3e05358 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -62,6 +62,15 @@ fi
 setprop sys.usb.ffs.aio_compat true
 setprop persist.adb.nonblocking_ffs false
 
+# Redmi Note 9S/Pro/Max props
+if getprop ro.vendor.build.fingerprint |grep -qi -e redmi/curtana -e redmi/joyeuse -e redmi/excalibur; then
+    setprop ro.surface_flinger.has_HDR_display true
+    setprop ro.surface_flinger.has_wide_color_display true
+    setprop persist.device_config.runtime_native.usap_pool_enabled true
+    setprop debug.sf.latch_unsignaled 1
+    setprop debug.sf.enable_gl_backpressure 1
+fi
+
 fixSPL() {
     if [ "$(getprop ro.product.cpu.abi)" = "armeabi-v7a" ]; then
         setprop ro.keymaster.mod 'AOSP on ARM32'
@@ -431,7 +440,7 @@ if getprop ro.vendor.build.fingerprint | grep -iq \
     -e xiaomi/grus -e xiaomi/cereus -e xiaomi/cactus -e xiaomi/raphael -e xiaomi/davinci \
     -e xiaomi/ginkgo -e xiaomi/willow -e xiaomi/laurel_sprout -e xiaomi/andromeda \
     -e iaomi/renoir \
-    -e redmi/curtana -e redmi/picasso \
+    -e redmi/curtana -e -e redmi/joyeuse -e redmi/excalibur -e redmi/picasso \
     -e bq/Aquaris_M10 ; then
     mount -o bind /mnt/phh/empty_dir /vendor/lib64/soundfx
     mount -o bind /mnt/phh/empty_dir /vendor/lib/soundfx
@@ -948,7 +957,7 @@ if getprop ro.vendor.build.fingerprint |grep -iq xiaomi/cepheus;then
     setprop ro.netflix.bsp_rev Q855-16947-1
 fi
 
-if getprop ro.vendor.build.fingerprint |grep -qi redmi/curtana;then
+if getprop ro.vendor.build.fingerprint |grep -qi redmi/curtana -e redmi/joyeuse -e redmi/excalibur; then
     setprop ro.netflix.bsp_rev Q6250-19132-1
 fi
 
-- 
2.25.1

